src/vendors/reducer.ts

```diff
@@ -0,0 +1,31 @@
+import { Actions, ActionTypes } from 'src/vendors/actions'
+import { Vendor } from 'src/vendors/types'
+
+export interface State {
+  vendors: { [name: string]: Vendor | null }
+  loading: boolean
+}
+
+export const initialState = {
+  vendors: {},
```

Comment:
Let's not create a property with the same name as its namespace. This is pedantic but probably a best practice so that we can always refer to these things specifically and accurately in future conversations. Suggestion `allVendors` or up to your discretion.